modules/ha-cluster/variables.tf

```diff
@@ -0,0 +1,328 @@
+}
+
+variable "aws_region" {
+  description = "What region the cluster and resources will be deployed into."
+  type        = string
+  default     = "us-east-1"
+}
+
+variable "vpc_id" {
+  description = "The ID of the existing VPC."
+  type        = string
+}
+
+variable "private_subnet_ids" {
+  description = "The existing VPC Private Subnets. Should be at least 3 and should match server count."
+  type        = list(string)
+}
+
+variable "public_subnet_ids" {
+  description = "The existing VPC Public Subnets. Should be at least 3 and should match server count."
+  type        = list(string)
+}
+
+variable "consul_image" {
+  type        = string
+  description = "Name of the Consul Agent Docker Image"
+  default     = "public.ecr.aws/hashicorp/consul:1.12.2"
+}
+
+variable "operating_system_family" {
+  default = "LINUX"
+}
+
+variable "cpu_architecture" {
+  default = "X86_64"
+}
+
+variable "docker_username" {
+  type        = string
+  description = "Username for Docker Hub authentication."
+  default     = null
+}
+
+variable "docker_password" {
+  type        = string
+  description = "Password (or token) for Docker Hub authentication."
+  default     = null
+}
+
+variable "lb_enabled" {
+  description = "Whether to create an ALB for the server task. Useful for accessing the UI."
+  type        = bool
+  default     = true
+}
+
+variable "internal_alb_listener" {
+  type        = bool
+  description = "ALB is internal-only to reduce load-test costs. When false the ALB will be accessible over the public network."
+  default     = true
+}
+
+variable "deploy_efs_cluster" {
+  type        = bool
+  description = "Deploy EFS Cluster for Consul data storage? Default is true."
+  default     = true
+}
+
+variable "ecs_cluster_name" {
+  description = "Specify an ECS cluster name to deploy the consul services."
+  default     = null
+}
+
+variable "create_ecs_log_group" {
+  description = "Create an ECS Log Group for the containers. Default is True"
+  default     = true
+}
+
+variable "ecs_log_retention_period" {
+  description = "Specify a log retnetion period in days. Default is 7."
+  default     = 7
+}
+
+variable "consul_container_cpu" {
+  description = "Set the Consul Container total CPU limit. Default is 2048."
+  default     = 2048
+}
+
+variable "consul_container_memory" {
+  description = "Set the Consul Cotainer total Memory limit. Default is 4096."
+  default     = 4096
+}
+
+variable "consul_task_cpu" {
+  description = "Set the Consul Server task CPU limit. Default is 1792."
+  default     = 1792
+}
+
+variable "consul_task_memory" {
+  description = "Set the Consul Server container Memory limit. Default is 3584."
+  default     = 3584
+}
+
+variable "datadog_task_cpu" {
+  description = "Set the Consul Server task CPU limit. Default is 1792."
+  default     = 256
+}
+
+variable "datadog_task_memory" {
+  description = "Set the Consul Server container Memory limit. Default is 3584."
+  default     = 512
+}
```

Comment:
Please remove the datadog specific functionality.

Instead, we can support generic variables, like `additional_container_definitions`, `additional_task_role_policies`, etc, in order to inject an additional container into the task definition. That way we can support any kind of additional "sidecar" container.